### PR TITLE
Updated "SupertypeAggregateCore" to attach Display names. Fixes #549

### LINKED ISSUE #549: Display Name handling in If() statement,
Display Names information doesn't seem to propagate through an If() statement. 

`r1.DisplayName` binds, 

but wrapping it in an If() loses the display name (note both branches are the same type)
`If(true,r1,r1).Display1`
Gets the error:
{Error 14-23: Name isn't valid. 'Display1' isn't recognized.}

If this is actually by design, we should give a better error message. 

Consider this test 

        [Fact]
        public void DisplayName2()
        {
            var r1 = RecordType.Empty()
                        .Add(new NamedFormulaType("F1", FormulaType.Number, "Display1"))
                        .Add(new NamedFormulaType("F0", FormulaType.Number, "Display0")); // Common

            var parameters = RecordType.Empty()
                .Add("r1", r1);

            var engine = new Engine(new PowerFxConfig());

            var result = engine.Check("r1.Display1", parameters);
            Assert.True(result.IsSuccess); // Succeeds

            // If takes the subset, common fields from each. 
            result = engine.Check("If(true,r1,r1).Display1", parameters); // !!!   Fails!!!!
            Assert.True(result.IsSuccess);
        }

Related issues:
- What happens when there are different types like `If(test,r1,r2).Display` . Can we trick Display to bind to different fields on r1,r2? 
- At the SDK level, should RecordValue.GetField() handle display names?

### PR DESCRIPTION
**Pull request by jas-valgotar**
- Updated "SupertypeAggregateCore" to fix display name issue with Aggregate types. (Fixes #549)
 - Added tests in "DisplayNameTest" for validation.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs [hunk 1]
```diff
@@ -263,5 +263,40 @@ public void ValidateExpressionConversionCommaSeparatedLocale(string inputExpress
...
+        // If types are different, you have no access to Display name.
+        [InlineData("If(true, r1, r2).Display1", false)]
+        [InlineData("If(true, r1, r2).Display0", false)]
+        [InlineData("If(true, r1, {Display1 : 123}).Display1", false)]
+
+        // If types are different, you have access to logical name, only if the name and type are same!
+        [InlineData("If(true, r1, r2).F1", true)]
+        [InlineData("If(false, r1, r2).F1", true)]
+        [InlineData("If(true, r1, r2).F0", false)]
+        public void DisplayNameTest(string input, bool succeeds)
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs:282]: This is a great test!

Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs:282]: Great tests!

Comment (jas-valgotar) [reply src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs:282]: Thanks Mike! :)

File: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs [hunk 2]
```diff
@@ -2345,7 +2345,12 @@ private static DType SupertypeAggregateCore(DType type1, DType type2, bool useLe
             }
 
             Contracts.Assert(!fError);
-            return new DType(type1.Kind, treeRes, UnionDataSourceInfoMetadata(type1, type2));
+            var returnType = new DType(type1.Kind, treeRes, UnionDataSourceInfoMetadata(type1, type2), type1.DisplayNameProvider);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:2348]: Wondering ... are there other similar bugs here for other type algebra operations? 
 Ie, any place that we call UnionDataSourceInfoMetadata without also checking DisplayNameProvider?

**Review by MikeStall — APPROVED** [review]

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**